arch/arm/core/aarch64/reset.c

```diff
@@ -29,9 +29,11 @@ void __weak z_arm64_el1_plat_init(void)
 
 void z_arm64_el_highest_init(void)
 {
-	write_cntfrq_el0(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	if (is_el_highest()) {
+		write_cntfrq_el0(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
 
-	z_arm64_el_highest_plat_init();
+		z_arm64_el_highest_plat_init();
```

Comment:
You want to execute `z_arm64_el_highest_plat_init()` always, not only when the `EL` you are entering to is actually the highest EL implemented.

Probably you are right that the name is confusing but the principle is:
- `z_arm64_el_highest_init()` and `z_arm64_el_highest_plat_init()` are executed at the highest level you enter Zephyr (that can or cannot be coincident with the highest EL implemented)
- `is_el_highest()` is checking if the highest EL __implemented__ is the current one

What do you think about changing the name `is_el_highest` to `is_el_highest_implemented` for the sake of clarity?

Reply:
your idea is good. Fix the code in next version.